### ISSUE TITLE
How to add self-made annoDb?

### ISSUE THREAD

**Issue by xlhu66** [tzhu-bio/cisDynet#7]
Dear Tao,
I am wondering how can we add the self-made annoDb for peak annotations? It is common that people do not study on the model species. As I know, specific TxDb can be generated in ChIPseeker, I am not sure if it can be achieved in cisDynet. Look forward to your reply. Thanks.

xiaoli

**Comment by tzhu-bio** [comment]
Here I am using ChIPSeeker to do the peaking annotations.

You can build your own annotation package with [AnnotationForge](https://bioconductor.org/packages/release/bioc/html/AnnotationForge.html).

**Comment by xlhu66** [comment]
Dear Tao, 
Sorry to make you confused. What I mean is shown in the below.
![屏幕截图 2024-01-12](https://github.com/tzhu-bio/cisDynet/assets/155941706/d954e08f-bed1-4cd0-a196-cd6f35b5aac6)

**Comment by tzhu-bio** [comment]
I understand what you mean, you can try to do it with the package I mentioned above.

**Comment by xlhu66** [comment]
Sorry, that is my mistake, what I mean is the file of gene structure annotation, which is same as  TxDb in ChiPseeker.

**Comment by tzhu-bio** [comment]
The gff file should be the most basic file. If the species you are studying has a genome sequence, it should have its corresponding gff annotation file. 

If you have provided the gff file in _**[addAnnotation](https://tzhu-bio.github.io/cisDynet_bookdown/book/getting-started-with-cisdynet.html#getting-set-up)**_, you can try to run this step directly.

**Comment by xlhu66** [comment]
> The gff file should be the most basic file. If the species you are studying has a genome sequence, it should have its corresponding gff annotation file.
> 
> If you have provided the gff file in _**[addAnnotation](https://tzhu-bio.github.io/cisDynet_bookdown/book/getting-started-with-cisdynet.html#getting-set-up)**_, you can try to run this step directly.

I see, I definitely forget to run these basic codes at the beginning. Is it ok to run addAnnotation right now after finish runing the Snakemake Pipeline already.  Or I have to rerun all codes orderly??

**Comment by tzhu-bio** [comment]
If no error is reported, you can not re-run it.
If cisDynet is indeed helpful for your research, it is recommended to read through it, some of the issues were explained when I wrote the documentation.

**Comment by xlhu66** [comment]
> If no error is reported, you can not re-run it. If cisDynet is indeed helpful for your research, it is recommended to read through it, some of the issues were explained when I wrote the documentation.

That is true. I will read the documentation carefully when I get any errors. Thank you so much.

**Comment by xlhu66** [comment]
Dear Tao,

After running addAnnotation, and building annoDb, I still got the error: 

Error in checkGeAnno() :
  Necessary Annotation not exist. Please **addAnnotation** first.
Calls: annoPeaks -> checkGeAnno
Execution halted

**Comment by tzhu-bio** [comment]
Check that your file format matches that of the example and make sure the path is correct.

**Comment by xlhu66** [comment]
> Check that your file format matches that of the example and make sure the path is correct.

I delete the comment, Sure, thanks
